src/sciline/domain.py

```diff
@@ -3,36 +3,55 @@
 from typing import Any, Generic, TypeVar, get_args, get_origin
 
 PARAM = TypeVar("PARAM")
+PARAM2 = TypeVar("PARAM2")
 SUPER = TypeVar("SUPER")
 
 
+def _check_supertype(cls: type) -> None:
+    # Mypy does not support __orig_bases__ yet(?)
+    # See also https://stackoverflow.com/a/73746554 for useful info
+    scope = cls.__orig_bases__[0]  # type: ignore[attr-defined]
+    # Only check direct subclasses
+    if get_origin(scope) is Scope:
+        supertype = get_args(scope)[1]
```

Comment:
Shouldn't it be the last one of the type arguments?

```python
supertype = get_args(scope)[-1]
```

Reply:
Good catch! Also the line above was wrong. Added a test.